device/src/region/fixed_channel_plans/mod.rs

```diff
@@ -8,15 +8,71 @@ mod us915;
 pub(crate) use au915::AU915;
 pub(crate) use us915::US915;
 
+#[derive(Clone)]
+struct PreferredJoinChannels {
+    channel_list: heapless::Vec<u8, 16>,
```

Comment:
It seems like your use of ChannelMask from the previous iteration has been abandoned in favor of this? Why may I only have 16 preferred channels?

I think that if you scope this back down to `fixed_channel_plans/mod.rs` as I mentioned elsewhere, it might be easy to implement use `ChannelMask<9>` here. You could create some constructors that assist in creating ChannelMasks that correspond to what users may typically do, such as `new_with_subband`.

One potential benefit of this approach too is that our stack could potentially "learn" a favorite subband upon Join and use that in future Joins (with default retries = 1). It's perhaps best to leave that feature for another PR, but I do think that speaking of "preferred subband" in terms of a "channel mask" has benefits in that regard as well.

Reply:
I made a 16 channel limit just in the spirit of saving memory. There's no technical reason, but I figured I could easily change it back to 72 if needed. It's just that usually gateways will have a max of 16 channels enabled (actually 18 counting 500 kHz channels).

One reservation I have about this is I don't find the `ChannelMask` API super intuitive. It's more oriented towards channel banks rather than individual channels. I definitely understand why it's designed that way, but it makes it harder to use in non-MAC contexts.

 One of my goals with this PR is to keep the number of RNG calls bounded, in preparation for if #109 gets accepted - if that is a feature people want, obviously. That's done much more easily if we hold an array of enabled channels, instead of looping randomly through all channels until we find one that is enabled. 
However now that I think of it, maybe temporarily collecting enabled channels into a `Vec` when needed would do the trick.

Reply:
> I made a 16 channel limit just in the spirit of saving memory.

`heapless::Vec<u8, 16>` is at least 16 bytes. `ChannelMask<9>` is 9 bytes.

> One reservation I have about this is I don't find the ChannelMask API super intuitive. 

I agree, but that's what I recommend we extend and work on rather than creating a novel type which serves the same function. Two types that functionally do the same thing doesn't sit well with me, personally.

Reply:
> One of my goals with this PR is to keep the number of RNG calls bounded. [...] That's done much more easily if we hold an array of enabled channels, instead of looping randomly through all channels until we find one that is enabled.

Your solution to minimizing RNG calls creates additional RAM overhead as one has to store a channel list for indexing instead of using the ChannelMask. I would isolate that kind of change to its own PR since that goal adopts an alternate design trade-off than what is currently adopted in this repository. 

Up until now, my understanding is that we should prioritize minimizing RAM usage, since RAM is generally the most precious resource on micro-controllers (as opposed to using a few more instructions to generate additional entropy).

Reply:
@jbeaurivage [I've added a PR](https://github.com/ivajloip/rust-lorawan/pull/114) that I think will completes the function set needed for manipulating ChannelMask.

I believe an API like this would be a good start:

```rust 
enum Subband {
   _1,
   ...
  _8,
}
impl<const D: usize, F: FixedChannelRegion<D>> FixedChannelPlan<D, F> {
    fn set_preferred_join_subband(subband: Subband) {
         // here you enable the appopriate bank of channels and then enable the appropriate fat channel using the new ChannelMask API
    }
}
```

This would cover 95% of these "preferred join channel" use-cases as networks tend to standardize on a specific subband. The use of the enum would maintain a safe public API. You could still provide a more dangerous `set_preferred_join_channel` API off of the FixedChannelPlan trait for special cases where users might have different needs :shrug: